### PR TITLE
remove href from a tag to prevent backbone page reload and change som…

### LINKED ISSUE #7893: issues with history sharing
this is from Main

I am unable to edit URL of a shared history

cannot-edit

js console has `GET https://usegalaxy.org/api/histories/undefined/sharing 400 (Bad Request)`

bonus: the url has an extra `:` in the link name
Galaxy

**extra extra bonus** get rid of the custom yellow 'pencil icon' and replace with standard font awesome

### PR DESCRIPTION
**Pull request by WilliamHolden**
fixes #7893

### CODE REVIEW

File: client/galaxy/scripts/components/Sharing.vue [hunk 1]
```diff
@@ -25,7 +25,7 @@
                         <span id="item-url-text" style="display: none">
                             {{ item_url_parts[0] }}<span id="item-identifier">{{ item_url_parts[1] }}</span>
                         </span>
-                        <a href="#" id="edit-identifier"><img :src="pencil_url"/></a>
+                        <a id="edit-identifier"><img :src="pencil_url" alt="Edit Share Url"/></a>
```
Comment (martenson) [inline client/galaxy/scripts/components/Sharing.vue]:
```suggestion
                        <a href="javascript:void(0)" id="edit-identifier"><img :src="pencil_url" alt="Edit Share Url"/></a>
```

Comment (martenson) [reply client/galaxy/scripts/components/Sharing.vue]: links that don't have href are often styled differently

Comment (WilliamHolden) [reply client/galaxy/scripts/components/Sharing.vue]: Good Point! Updated.

**Comment by martenson** [comment]
@WilliamHolden you can save some force pushing work next time by just clicking on the commit button next to my suggestion. Thanks for incorporating!

**Comment by blankenberg** [comment]
This should probably be cherry-picked to at least 19.05

**Pull request merged**